src/components/ui/Tiles/Tiles.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { forwardRef } from 'react'
+import type { HTMLAttributes, ReactElement } from 'react'
+
+import Tile from './Tile'
+
+export interface TilesProps extends HTMLAttributes<HTMLDivElement> {
+  /**
+   * ID to find this component in testing tools (e.g.: cypress,
+   * testing-library, and jest).
+   */
+  testId?: string
+}
+
+const MIN_CHILDREN = 2
+const MAX_CHILDREN = 4
+const NUMBER_ITEMS_TO_EXPAND = 3
+
+const Tiles = forwardRef<HTMLDivElement, TilesProps>(function Tiles(
+  { testId = 'store-tiles', children, ...otherProps },
+  ref
+) {
+  const childrenCount = React.Children.count(children)
+  const spanFirstChild = childrenCount === NUMBER_ITEMS_TO_EXPAND
+  const isOutOfBounds =
+    childrenCount < MIN_CHILDREN || childrenCount > MAX_CHILDREN
+
+  if (isOutOfBounds) {
+    throw new Error(
+      `Tiles cannot receive less than ${MIN_CHILDREN} or more than ${MAX_CHILDREN} children.`
+    )
+  }
+
+  React.Children.forEach(children as ReactElement, (child) => {
+    if (child.type !== Tile) {
+      throw new Error('Only Tile components allowed as children.')
+    }
```

Comment:
Does this error is being used to warn the developer about the bounds? What do you think about only warning on developing mode?
```suggestion
    if (process.env.NODE_ENV === 'development') {
      if (child.type !== Tile) {
        throw new error ('Only Tile components allowed as children.')
      }
    }
```

Reply:
Hmm, not sure. Since we don't want to skip the validation in the other environments like `staging` or `production`.
I also think is a little weird to check the environment inside the component. What do you think?

Reply:
I think environment checks should be the closest as possible to the request handler.

The component raises the error, whoever calls this component should be responsible to handle (or raise) the error and, if necessary, applying some environment check.

Reply:
Some references : [mui](https://github.com/mui-org/material-ui/blob/1230b1a571942f1abf2fec81812b757ceb0e5d0b/packages/mui-system/src/spacing.js#L104), [reakit](https://github.com/reakit/reakit/blob/a211d94da9f3b683182568a56479b91afb1b85ae/packages/reakit-warning/src/warning.ts#L13).